modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreator.java

```diff
@@ -418,4 +435,27 @@ else if (value.charAt(start) == 'Z')
       return tz;
    }
 
+   /**
+    * Get UsernameToken profile digest
+    *
+    * @param nonce nonce
+    * @param created creation timestamp
+    * @param password clear text password
+    * @return Password_Digest = Base64 ( SHA-1 ( nonce + created + password ) )
+    */
+   private static String getUsernameTokenPasswordDigest(String nonce, String created, String password) {
+      ByteBuffer buf = ByteBuffer.allocate(1000);
+      buf.put(Base64.getDecoder().decode(nonce));
+      try {
+         buf.put(created.getBytes("UTF-8"));
+         buf.put(password.getBytes("UTF-8"));
+      } catch (UnsupportedEncodingException e) {
+         e.printStackTrace();
```

Comment:
@Skyllarr  Is it good to add some log message here ?

Reply:
@jimma Fixed